source/PluginDev/Assets/GooglePlayGames/Platforms/Android/AndroidRealTimeMultiplayerClient.cs

```diff
@@ -34,111 +56,689 @@ public AndroidRealTimeMultiplayerClient(AndroidJavaObject account)
-                    roomConfigBuilder.Call<AndroidJavaObject>("setAutoMatchCriteria",
-                        roomConfigBuilder.CallStatic<AndroidJavaObject>("createAutoMatchCriteria", (int) minOpponents,
-                            (int) maxOpponents, (long) exclusiveBitMask));
+                    if (variant > 0)
+                    {
+                        roomConfigBuilder.Call<AndroidJavaObject>("setVariant", (int) variant);
+                    }
+
+                    using (var autoMatchCriteria = roomConfigClass.CallStatic<AndroidJavaObject>(
+                        "createAutoMatchCriteria", (int) minOpponents,
+                        (int) maxOpponents, (long) exclusiveBitMask))
+                    using (roomConfigBuilder.Call<AndroidJavaObject>("setAutoMatchCriteria", autoMatchCriteria))
+                        ;
+
+                    using (var messageReceivedListener = new AndroidJavaObject(
+                        "com.google.games.bridge.RealTimeMessageReceivedListenerProxy",
+                        new MessageReceivedListenerProxy(listener)))
+                    using (roomConfigBuilder.Call<AndroidJavaObject>("setOnMessageReceivedListener",
+                        messageReceivedListener))
+                        ;
+
+                    using (var roomStatusUpdateCallback = new AndroidJavaObject(
+                        "com.google.games.bridge.RoomStatusUpdateCallbackProxy",
+                        new RoomStatusUpdateCallbackProxy(this, listener)))
+                    using (roomConfigBuilder.Call<AndroidJavaObject>("setRoomStatusUpdateCallback",
+                        roomStatusUpdateCallback))
+                        ;
+
                     mRoomConfig = roomConfigBuilder.Call<AndroidJavaObject>("build");
+                    mListener = listener;
                 }
-            }
 
-            using (var task = mClient.Call<AndroidJavaObject>("create", mRoomConfig))
-            {
+                mMinPlayersToStart = (int) minOpponents + 1;
+
+                using (var task = mRtmpClient.Call<AndroidJavaObject>("create", mRoomConfig))
+                {
+                    AndroidTaskUtils.AddOnFailureListener(
+                        task,
+                        e =>
+                        {
+                            listener.OnRoomConnected(false);
+                            CleanSession();
+                        });
+                }
             }
         }
 
-        public void CreateWithInvitationScreen(uint minOpponents, uint maxOppponents, uint variant,
+        public void CreateWithInvitationScreen(uint minOpponents, uint maxOpponents, uint variant,
             RealTimeMultiplayerListener listener)
         {
-            // Task<Intent> getSelectOpponentsIntent(@IntRange(from = 1) int minPlayers, @IntRange(from = 1) int maxPlayers, boolean allowAutomatch)
+            AndroidHelperFragment.ShowRtmpSelectOpponentsUI(minOpponents, maxOpponents,
```

Comment:
Should we check if current room state is active and do nothing in this case?

Reply:
Yes we should. I added a control and get them in a lock block.